plugins/puma/puma_

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+# Copyright (c) 2016 Alexey Anikanov (alexey.anikanov@gmail.com)
+# License GPLv2
+# This plugin monitors number of workers, total memory used and average memory per process for puma,
+# a ruby web server built for concurrency http://puma.io
+# Here are the symlinks to enable it
+#
+#        ln -s /usr/share/munin/plugins/puma_ /etc/munin/plugins/puma_average
+#        ln -s /usr/share/munin/plugins/puma_ /etc/munin/plugins/puma_memory
+#        ln -s /usr/share/munin/plugins/puma_ /etc/munin/plugins/puma_processes
+
+mode=`echo $0 | cut -d _ -f 2`
+
+if [ "$1" = "suggest" ]; then
+        echo "memory"
+        echo "processes"
+        echo "average"
+        exit 0
+fi
+
+if [ "$mode" = "memory" ]; then
+        if [ "$1" = "config" ]; then
+                echo "graph_title Total puma Memory"
+                echo "graph_vlabel Total RAM"
+                echo "graph_category puma"
+                echo "graph_args --base 1024"
+                echo "ram.label Total RAM"
+                exit 0
+        else
+
+        memory_array=(`ps auwx | egrep "puma.* worker" | grep -v grep | awk '{print $6 }'`)
+
+        for i in "${memory_array[@]}"
+            do
+                 sum=$(( $sum + ( $i * 1024) ))
+        done
+                 echo -n "ram.value "
+                 echo $sum
+
+        fi
+
+elif [ "$mode" = "processes" ]; then
+        if [ "$1" = "config" ]; then
+                echo "graph_title puma Processes"
+                echo "graph_vlabel Processes"
+                echo "graph_category puma"
+                echo "processes.label active processes"
+        else
+                echo -n "processes.value "
```

Comment:
`echo -n` can be replaced with a `printf` which is much more standard when you don't want to emit a trailing new line char.

That said, your usual use of `eval` is perfectly sane : only use `printf` when a arg-less `echo` isn't enough.